presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -243,8 +243,11 @@ public void close()
     {
         try (Closer closer = Closer.create()) {
             closer.register(super::close);
-            if (credentialsProvider instanceof Closeable) {
-                closer.register((Closeable) credentialsProvider);
+            if (credentialsProvider.isPresent()) {
```

Comment:
@findepi has a good point about not needing `Optional`. But a general FYI, you can simplify code like this using `ifPresent`:
```java
credentialsProvider.ifPresent(provider -> {
    if (provider instanceof Closeable) { ... }
});

Reply:
Yep. Thanks.